lib/waterfoul/boot_rom.rb

```diff
@@ -49,10 +49,19 @@ class BootROM
       0x3E, 0x01, 0xE0, 0x50
     ].freeze
 
+    attr_reader :cartridge
```

Comment:
I thought it's a nice way to model how the memory lookup works while booting: low addresses are overridden by the boot rom, but higher addresses access the real rom. So this "chains" them nicer than two `if` in MMU.
Also, this allows just replacing the rom by the real cartridge once boot is finished, instead of maintaining an extra `@map_boot_rom` state in MMU.